k8s/src/main/scala/io/buoyant/k8s/EndpointsNamer.scala

```diff
@@ -121,9 +125,12 @@ abstract class EndpointsNamer(
   labelName: Option[String] = None,
   backoff: Stream[Duration] = EndpointsNamer.DefaultBackoff
 )(implicit timer: Timer = DefaultTimer)
-  extends Namer {
+  extends Namer with Admin.WithHandlers {
   import EndpointsNamer._
 
+  // Metadata about the state of portMap watches.
+  private[this] val portMapInstrumentation = mutable.Map[(String, String, Option[String]), (InstrumentedActivity[NumberedPortMap], WatchState[v1.Service, v1.ServiceWatch])]()
```

Comment:
This value looks like it contains a lot of information that isn't easily recognizable at first glance. Do you think it would be helpful for this to be a class of some sort?